### ISSUE TITLE
Rails 5.1 compatibility

### ISSUE THREAD

**Issue by voxik** [marcandre/backports#114]
Testing backports against activesupport 5.1, I observe following test suite error:

~~~
Failure: test_rails(AAA_TestBackportGuards):
  <{Array=>[".alias_method_chain"],
   Binding=>[".alias_method_chain"],
   Dir=>[".alias_method_chain"],
   Enumerable=>[".alias_method_chain"],
   Integer=>[".alias_method_chain"],
   Float=>[".alias_method_chain"],
   GC=>[".alias_method_chain"],
   Hash=>[".alias_method_chain"],
   IO=>[".alias_method_chain"],
   Kernel=>[".alias_method_chain"],
   Math=>[".alias_method_chain"],
   MatchData=>[".alias_method_chain"],
   Method=>[".alias_method_chain"],
   Module=>[:alias_method_chain, ".alias_method_chain"],
   Numeric=>[".alias_method_chain"],
   ObjectSpace=>[".alias_method_chain"],
   Proc=>[".alias_method_chain"],
   Process=>[".alias_method_chain"],
   Range=>[".alias_method_chain"],
   Regexp=>[".alias_method_chain"],
   String=>[".alias_method_chain"],
   Struct=>[".alias_method_chain"],
   Symbol=>[".alias_method_chain"],
   #<Class:#<Object:0x0000558bf763c0a8>>=>[".alias_method_chain"],
   #<Class:#<ARGF.class:0x0000558bf7633b38>>=>[".alias_method_chain"],
   Enumerator=>[".alias_method_chain"]}> was expected to be nil.
/builddir/build/BUILD/backports-3.7.0/usr/share/gems/gems/backports-3.7.0/test/_backport_guards_test.rb:148:in `test_rails'
     145:     before = digest
     146:     require "backports/rails"
     147:     after = digest
  => 148:     assert_nil digest_delta(before, after)
     149:   end
     150: end
~~~

It seems that his happens, because the alias_method_chaing method was removed in Rails 5.1 (https://github.com/rails/rails/pull/27035/commits/7c848e6dd493ff236d33a0410a92f4c3e5cc3c7f)

This helps as a workaround:

~~~
sed -i "s/^/#/" lib/backports/rails/module.rb
~~~

But you probably want some fancier solution. Not sure.

**Comment by marcandre** [comment]
Thanks for the great report.

I'll probably just lock the version of AS I'm testing against.

**Comment by marcandre** [comment]
I'm not sure how this was tested. The Gemfile specifies `~3.2.0`...

Note that `activesupport` is only used for testing.

**Comment by voxik** [comment]
This was tested without Bundled indeed. I thought that it should test Rails support and compatibility, so it should work with any version of Rails, but I might be wrong. In that case feel free to close the ticket.

**Comment by voxik** [comment]
At the end, this was resolved by #127.
